### PR TITLE
Qerrormessage bug fix

### PR DESCRIPTION
**Pull request by BobleChinois**
In the https://github.com/lightning-power-users/node-launcher/commit/9c8726f5e1388e6a11f8b26706ddae3e43b3638c commit, I modified the 2 following files : 
* [lnd_wallet_layout.py](https://github.com/lightning-power-users/node-launcher/compare/master...BobleChinois:qerrormessage_bug_fix?expand=1#diff-4fd8dd6257b0e8a441b17d1a5d6315e8): replaced the faulty line with `self.error_message.showMessage(details)`
* [test_lnd_wallet_layout.py](https://github.com/lightning-power-users/node-launcher/compare/master...BobleChinois:qerrormessage_bug_fix?expand=1#diff-4c12f1c6d15e52be50b7417e3183ca09): this is the unit test you wrote, I didn't make any change.

I tried run the tests, and found out the new test will generate the following error:
```
self = <test_lnd_wallet_layout.TestLndWalletLayout object at 0x7f4134474eb8>
seed_dialog_patch = <MagicMock name='SeedDialog' id='139918026653312'>
thread_pool_patch = <MagicMock name='QThreadPool' id='139918026702576'>
input_dialog_patch = <MagicMock name='QInputDialog' id='139918026710936'>
error_message_patch = <MagicMock name='QErrorMessage' id='139918026743880'>
lnd_client_patch = <MagicMock name='LndClient' id='139918026752296'>
keyring_patch = <MagicMock name='keyring' id='139918026764752'>
lnd_wallet_layout = <node_launcher.gui.network_buttons.lnd_wallet_layout.LndWalletLayout object at 0x7f4134460808>
qtbot = <pytestqt.qtbot.QtBot object at 0x7f413446bef0>

    def test_handle_lnd_poll_error_message(self,
                           seed_dialog_patch: MagicMock,
                           thread_pool_patch: MagicMock,
                           input_dialog_patch: MagicMock,
                           error_message_patch: MagicMock,
                           lnd_client_patch: MagicMock,
                           keyring_patch: MagicMock,
                           lnd_wallet_layout: LndWalletLayout,
                           qtbot: QTest):
        lnd_wallet_layout.handle_lnd_poll('unknown gRPC error detail')
>       error_message_patch.showMessage.assert_called()

tests/test_gui/test_network_buttons/test_lnd_wallet_layout.py:35: 
```
I'm not too sure what's wrong, but in commit https://github.com/lightning-power-users/node-launcher/commit/ace62e7361834b99733a9994f2d905de7ef00f47 I tried using the solution you proposed first `QErrorMessage.showMessage(self.error_message, details)`, and there's no error in the unit test. 
Except if you have an idea of what was wrong and want to try something else, if we agree to go with the fix as it is now I can add a third commit just to remove the commented out line and we will be ready to merge.

### CODE REVIEW

**Comment by PierreRochard** [comment]
Thank you @BobleChinois! I've had a couple more users encounter this bug this week.

The unit test was failing because we're patching the `QErrorMessage` class with a `MagicMock` object,  so `error_message_patch.showMessage.assert_called` is asserting that `showMessage` was called as a class method, while `lnd_wallet_layout.error_message.showMessage.assert_called` would assert that the instance method is being called. Let me know if that change doesn't make the test pass and I'll take a closer look

**Comment by justinmoon** [comment]
This test passes for me ...

**Comment by BobleChinois** [comment]
Ok just to clarify what's the issue here, because I realise it is a bit confusing and I wasn't very clear yesterday evening (sorry Justin!): 

- I first made the 9c8726f5e1388e6a11f8b26706ddae3e43b3638c commit, which **doesn't** pass test

- I then made the ace62e7361834b99733a9994f2d905de7ef00f47 where I only modified the lnd_wallet_layout.py file with a slightly different bug fix, that **does** pass the test (Justin it's the one you tested yesterday)

- Pierre then suggested to modify the test using the following line `lnd_wallet_layout.error_message.showMessage.assert_called()` instead of `error_message_patch.showMessage.assert_called()`, and see if the bug fix as is was first proposed in 9c8726f5e1388e6a11f8b26706ddae3e43b3638c now pass the test. It **doesn't**.

To sum it up, while we could just merge ace62e7361834b99733a9994f2d905de7ef00f47 and be fine, we are trying to understand here why 9c8726f5e1388e6a11f8b26706ddae3e43b3638c is failing the test while it seems to be a correct fix afaict.

Last commit 2bc25f73f3b3e20b39d1a112da51f259655c98b7 is what I'm trying to do following Pierre's last suggestion, but it fails and I don't understand why.

I hope it's clearer now, thank you very much you two for your help, it might seem a bit futile I guess but it has been really useful for me to understand how tests work!

Ho, and wish you a happy chinese new year of the pig!:pig2:

**Comment by PierreRochard** [comment]
Thank you for the explanation @BobleChinois! I merged it and will do a follow up PR if there's a problem with this one

**Comment by PierreRochard** [comment]
The mocking of the QErrorMessage class was not working, so I manually replaced the property with a mock object in the test fixture.

https://github.com/lightning-power-users/node-launcher/pull/161

**Comment by BobleChinois** [comment]
Thanks Pierre, so the issue was with the fixture... I'll get back to it when I'll have some time and nail it.
